share/eds/byzantine/bad_encoding.go

```diff
@@ -118,45 +118,42 @@ func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
 	if header.Height() != int64(p.BlockHeight) {
 		return errors.New("fraud: incorrect block height")
 	}
-	merkleRowRoots := header.DAH.RowRoots
-	merkleColRoots := header.DAH.ColumnRoots
-	if len(merkleRowRoots) != len(merkleColRoots) {
+
+	if len(header.DAH.RowRoots) != len(header.DAH.ColumnRoots) {
 		// NOTE: This should never happen as callers of this method should not feed it with a
 		// malformed extended header.
 		panic(fmt.Sprintf(
 			"fraud: invalid extended header: length of row and column roots do not match. (rowRoots=%d) (colRoots=%d)",
-			len(merkleRowRoots),
-			len(merkleColRoots)),
+			len(header.DAH.RowRoots),
+			len(header.DAH.ColumnRoots)),
 		)
 	}
-	if int(p.Index) >= len(merkleRowRoots) {
-		return fmt.Errorf("fraud: invalid proof: index out of bounds (%d >= %d)", int(p.Index), len(merkleRowRoots))
-	}
-	if len(merkleRowRoots) != len(p.Shares) {
-		return fmt.Errorf("fraud: invalid proof: incorrect number of shares %d != %d", len(p.Shares), len(merkleRowRoots))
-	}
 
-	root := merkleRowRoots[p.Index]
-	if p.Axis == rsmt2d.Col {
-		root = merkleColRoots[p.Index]
+	// change the order of dah roots as we are now collecting nmt proofs against orthogonal axis
+	roots := [][][]byte{header.DAH.ColumnRoots, header.DAH.RowRoots}
+	if int(p.Index) >= len(roots[rsmt2d.Row]) {
+		return fmt.Errorf("fraud: invalid proof: index out of bounds (%d >= %d)", int(p.Index), len(roots[rsmt2d.Row]))
```

Comment:
Let's mention that it's befp somehow in all the errrors.
Also, it's not fraud pkg anymore, I don't think we should prepend the byzantine here as well, just remove fraud

Reply:
https://github.com/celestiaorg/celestia-node/pull/2408/commits/68b161403000aae979a2bfd120fab69899c47c5f